File: PROCESSING/NatureOfCode/8_Fractals/RecursionCircle/src/Sketch.java
import processing.core.PApplet;

public class Sketch extends PApplet {

	public static void main(String[] args) {
		PApplet.main("Sketch");
	}
	
	
	void drawCircle(float x, float y, float radius) {
		ellipse(x, y, radius, radius);
		if (radius > 8) {
			drawCircle(x + radius/2, y, radius/2);
			drawCircle(x - radius/2, y, radius/2);
			drawCircle(x, y + radius/2, radius/2);
			drawCircle(x, y - radius/2, radius/2);
		}
	}
	

	@Override public void settings() {
		size(800, 800);
	}

	@Override public void setup() {
		background(255);
		stroke(0);
		drawCircle(width/2, height/2, 400);
	}

	@Override public void draw() {
		
	}
}

File: PROCESSING/junk/Stvorceky/src/Sketch.kt
import processing.core.PApplet

class Sketch : PApplet() {

	var minSize = 1f
	var i = 0f
	
	var x = 0f
	var y = 0f
	
	val ITERATIONS = 50

	override fun settings() {
		fullScreen()
	}

	override fun setup() {
		frameRate(20f)
		noFill()
		
		x = width/2f
		y = height/2f
	}

	override fun draw() {
		if (i > ITERATIONS) return
		
		stroke(0, (i/ITERATIONS)*255)
		ellipse(x - 50 + i*5, y - 50 + i*5, minSize/2, minSize/2)
		
		minSize *= 1.1f
		i += 0.5f
	}
	
}

fun main(args: Array<String>) {
	PApplet.main("Sketch")
}